### PR TITLE
Add documentation for cakeissues.net

### PR DESCRIPTION
**Pull request by pascalberger**
Add documentation for cakeissues.net, moved from Cake.Issues.Website repository including full history.

Part of https://github.com/cake-contrib/Cake.Issues.Website/issues/123

### CODE REVIEW

File: docs/examples.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        MsBuildIssuesFromFilePath(
+            logPath,
+            MsBuildXmlFileLoggerFormat),
+        repoRootFolder);
+
+    Information("{0} issues are found.", issues.Count());
+});
+```
+
+[MSBuild Extension Pack]: http://www.msbuildextensionpack.com/
```
Comment (x-jokay) [inline docs/examples.md]: [MD047](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md047---files-should-end-with-a-single-newline-character) markdown files should end with a single newline character.

File: docs/requirements.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+---
+Order: 10
+Title: Requirements
+Description: Requirements for the Cake.Issues.MsBuild addin.
+---
+The requirements for using the [Cake.Issues.MsBuild addin] are listed in the [release notes] for any specific version.
+
+[Cake.Issues.MsBuild addin]: https://www.nuget.org/packages/Cake.Issues.MsBuild
+[release notes]: release-notes
```
Comment (x-jokay) [inline docs/requirements.md]: [MD047](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md047---files-should-end-with-a-single-newline-character) markdown files should end with a single newline character.

File: docs/examples.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+```
+
+Finally you can define a task where you call the core addin with the desired issue provider.
+The following example reads issues reported as MsBuild warnings by the `XmlFileLogger`
+class from [MSBuild Extension Pack]:
+
+```csharp
+Task("Analyze-Log")
+.IsDependentOn("Build-Solution")
+.Does(() =>
```
Comment (x-jokay) [inline docs/examples.md]: Fix indention for `.IsDependentOn` and `.Does`.

File: docs/features.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+---
+The [Cake.Issues.MsBuild addin] provides the following features.
+
+# Basic features
+
+* Reads warnings from MSBuild log files.
+* Provides URLs for all code analysis (`CA*`) and StyleCop (`SA*`) warnings.
+* Support for custom URL resolving using the [MsBuildAddRuleUrlResolver] alias.
+
+# Supported log file formats
```
Comment (x-jokay) [inline docs/features.md]: [MD025](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md025---multiple-top-level-headings-in-the-same-document
) multiple top level headings in the same document.

File: docs/features.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+* [MsBuildBinaryLogFileFormat] alias for reading issues from binary log files.
+* [MsBuildXmlFileLoggerFormat] alias for reading issues from log files created by [MSBuild Extension Pack XmlFileLogger].
+
+# Supported comment formats
```
Comment (x-jokay) [inline docs/features.md]: [MD025](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md025---multiple-top-level-headings-in-the-same-document
) multiple top level headings in the same document.

File: docs/features.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+|                                                                    | Comment format                 | Remarks                        |
+|--------------------------------------------------------------------|--------------------------------|--------------------------------|
+| <span class="glyphicon glyphicon-ok" style="color:green"></span>   | `IssueCommentFormat.PlainText` |                                |
+| <span class="glyphicon glyphicon-remove" style="color:red"></span> | `IssueCommentFormat.Markdown`  |                                |
+| <span class="glyphicon glyphicon-remove" style="color:red"></span> | `IssueCommentFormat.Html`      |                                |
+
+# Supported IIssue properties
```
Comment (x-jokay) [inline docs/features.md]: [MD025](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md025---multiple-top-level-headings-in-the-same-document
) multiple top level headings in the same document.

File: docs/features.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+| <span class="glyphicon glyphicon-ok" style="color:green"></span> | `IIssue.ProviderName`             |                                |
+| <span class="glyphicon glyphicon-ok" style="color:green"></span> | `IIssue.ProjectName`              |                                |
+| <span class="glyphicon glyphicon-ok" style="color:green"></span> | `IIssue.ProjectFileRelativePath`  |                                |
+| <span class="glyphicon glyphicon-ok" style="color:green"></span> | `IIssue.AffectedFileRelativePath` |                                |
+| <span class="glyphicon glyphicon-ok" style="color:green"></span> | `IIssue.Line`                     |                                |
+| <span class="glyphicon glyphicon-ok" style="color:green"></span> | `IIssue.Message`                  |                                |
+| <span class="glyphicon glyphicon-ok" style="color:green"></span> | `IIssue.Priority`                 | Always [IssuePriority.Warning] |
+| <span class="glyphicon glyphicon-ok" style="color:green"></span> | `IIssue.PriorityName`             | Always [IssuePriority.Warning] |
```
Comment (x-jokay) [inline docs/features.md]: `Always [IssuePriority.Warning]` seems not correct for `IIssue.PriorityName`.

File: docs/examples.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+```csharp
+Task("Build-Solution").Does(() =>
+{
+    // Build solution.
+    var settings =
+        new MSBuildSettings()
+            .WithLogger(
+                Context.Tools.Resolve("MSBuild.ExtensionPack.Loggers.dll").FullPath,
+                "XmlFileLogger",
+                string.Format(
```
Comment (x-jokay) [inline docs/examples.md]: Add `CultureInfo.InvariantCulture`.

Comment (pascalberger) [reply docs/examples.md]: Changed to string interpolation

**Comment by pascalberger** [comment]
@x-jokay Ready for another review.

Top level headings are  required to have the entry appear in site TOC (See https://github.com/cake-contrib/Cake.Issues.Website/pull/41)

**Pull request merged**